### ISSUE TITLE
Customize PHP Docker via ENV variables

### ISSUE THREAD

**Issue by julianxhokaxhiu** [docker-library/php#308]
Hi,

I was looking at your Dockerfiles, scripts and surrounding, like Documentations. As far as I understood, to customize the PHP docker that I would like to target, I have to create my own Dockerfile.

Still I was looking if it's possible to customize it, without using a Dockerfile or Docker Compose. I'm asking this because I would not like to spawn my own Docker, have to link it in the main Hub in order to just rebuild it with just some extensions in it.

Is this in plan from your side, or is it just a no goal? Maybe do you have some script hooks that I could use every time I run the Docker? That also would be helpful.

Thank you in advance,
Julian

**Comment by schmunk42** [comment]
Related example: https://github.com/dmstr/docker-php-yii2/blob/253b8e154dda546535de3a8ecde89f8feb877471/php-7.0/image-files/usr/local/bin/docker-entrypoint.sh#L14

**Comment by zored** [comment]
I want to custumize `./configure` flags with `ENV`, for example. To create **FPM** + **ZTS** build. Should I create my own `Dockerfile` and rebuild PHP?

**Comment by yosifkit** [comment]
@zored, yeah zts and fpm are both items that need to be configured at build time. :cry:  There are many other modules that can be enabled afterward.  @schmunk42's example is a good solution if you want to enable them on start of your container.

**Comment by julianxhokaxhiu** [comment]
Although the example tackles one of the possible scenario, I sugget to provide at least three ENV variables which acts as:

- `PHP_PECL_EXTENSIONS`: a comma-separated list, of extension you would like to install from PECL ( since I saw we compile them from scratch and they don't come already pre-compiled, a possible issue is to find a way to auto-detect dev dependencies and install them via the package-manager )
- `PHP_EXTENSIONS`: a comma-separated list of native PHP extensions that come bundled with PHP src itself ( say `soap`, `xsl`, etc. )
- `APACHE_MODULES`: a comma-separated list, of Apache modules to be enabled ( say `rewrite` )

By providing these three simple ENV variables, you give everyone the freedom to use this docker without the need to create a custom one ( unless someone wants to do a more special customization ).

I hope I clarified the goal of this issue. Best regards.
